src/commands/createFunction/CSharpFunctionCreator.ts

```diff
@@ -29,7 +29,7 @@ export class CSharpFunctionCreator extends FunctionCreatorBase {
         this._ui = ui;
     }
 
-    public async promptForSettings(ui: IUserInterface, functionName: string | undefined, functionSettings: string[]): Promise<void> {
+    public async promptForSettings(ui: IUserInterface, functionName: string | undefined, functionSettings: {}): Promise<void> {
```

Comment:
Are functionSettings values just strings or any type?  Consider being more specific.

Also consider specifying known keys.

Reply:
Yeah it should be strings. I'll be more specific.

However, I don't plan on specifying known keys. There's over 20 templates and they each have different keys. It's up to user's of the api to know what keys to specify for the template they're using (which is good enough for now IMO)